bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkMutationGCJClient.java

```diff
@@ -0,0 +1,184 @@
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TestBulkMutationGCJClient {
+
+  @Rule
+  public ExpectedException expect = ExpectedException.none();
+
+  @Mock private UnaryCallable<RowMutation, Void> callable;
+  private BulkMutationGCJClient bulkMutationClient;
+  private SettableApiFuture<Void> future;
+  private RowMutation rowMutation = RowMutation.create("fake-table", "fake-key");
+  private final BulkMutationBatcher batcher = new BulkMutationBatcher(new BatcherUnaryCallable());
+
+  @Before
+  public void setUp() {
+    BulkMutationBatcher bulkMutationBatcher = new BulkMutationBatcher(callable);
+    bulkMutationClient = new BulkMutationGCJClient(bulkMutationBatcher);
+    future = SettableApiFuture.create();
+  }
+
+  @Test
+  public void testAdd() {
+    when(callable.futureCall(rowMutation)).thenReturn(future);
+    ApiFuture<Void> result = bulkMutationClient.add(rowMutation);
+    Assert.assertFalse(result.isDone());
+    future.set(null);
+    Assert.assertTrue(result.isDone());
+    verify(callable).futureCall(rowMutation);
+  }
+
+  @Test(expected = ExecutionException.class)
+  public void testAddFailure() throws Exception{
+    String message = "can not perform mutation";
+    when(callable.futureCall(rowMutation)).thenReturn(future);
+    future.setException(new RuntimeException(message));
+    ApiFuture<Void> result = bulkMutationClient.add(rowMutation);
+    Assert.assertTrue(result.isDone());
+    result.get();
+    expect.expectMessage(message);
+    verify(callable).futureCall(rowMutation);
+  }
+
+  @Test
+  public void testCallableTooFewStatuses() {
+    RowMutation rowMutationOther = RowMutation.create("table", "key");
+    SettableApiFuture<Void> futureOther = SettableApiFuture.create();
+    when(callable.futureCall(rowMutation)).thenReturn(future);
+    when(callable.futureCall(rowMutationOther)).thenReturn(futureOther);
+
+    ApiFuture<Void> result = bulkMutationClient.add(rowMutation);
+    ApiFuture<Void> resultOther = bulkMutationClient.add(rowMutationOther);
+    // Only resolving one response
+    future.set(null);
+
+    Assert.assertTrue(result.isDone());
+    Assert.assertFalse(resultOther.isDone());
+
+    Assert.assertFalse(bulkMutationClient.isFlushed());
+    futureOther.set(null);
+    Assert.assertTrue(bulkMutationClient.isFlushed());
+
+    verify(callable).futureCall(rowMutation);
+    verify(callable).futureCall(rowMutationOther);
+  }
+
+  @Test
+  public void testConcurrentBatches() throws Exception {
```

Comment:
What are you testing here?

Reply:
My intention was to test the list of future response which i was maintaining in earlier implementation.
 
Now that we have moved to `OperationAccountant` so we don't really need this test case.

Reply:
Have removed testConcurrentBatches as there are separate test cases for OperationAccountant.